includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -532,6 +558,128 @@ function( $scope ) {
 		exit();
 	}
 
+	/**
+	 * Determines whether the authentication proxy is used.
+	 *
+	 * In order to streamline the setup and authentication flow, the plugin uses a proxy mechanism based on an external
+	 * service. This can be overridden by providing actual GCP credentials with the {@see 'googlesitekit_oauth_secret'}
+	 * filter.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return bool True if proxy authentication is used, false otherwise.
+	 */
+	public function using_proxy() {
+		$credentials = $this->get_client_credentials();
+
+		// If no credentials yet, assume true.
+		if ( ! is_object( $credentials ) || empty( $credentials->web->client_id ) ) {
+			return true;
+		}
+
+		// If proxy credentials, return true.
+		if ( false !== strpos( $credentials->web->client_id, '.apps.sitekit.withgoogle.com' ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns the setup URL to the authentication proxy.
+	 *
+	 * @param string $code Optional. Temporary access code for an undelegated access token. Default empty string.
+	 * @return string URL to the setup page on the authentication proxy.
+	 */
+	public function get_proxy_setup_url( $code = '' ) {
+		$url = self::PROXY_URL . '/site-management/setup/';
+
+		$credentials = $this->get_client_credentials();
+
+		if ( ! is_object( $credentials ) || empty( $credentials->web->client_id ) ) {
+			$home_url           = home_url();
+			$home_url_no_scheme = str_replace( array( 'http://', 'https://' ), '', $home_url );
+
+			$rest_root = str_replace( array( 'http://', 'https://' ), '', rest_url() );
+			$rest_root = str_replace( $home_url_no_scheme, '', $rest_root );
+
+			$admin_root = str_replace( array( 'http://', 'https://' ), '', admin_url() );
+			$admin_root = str_replace( $home_url_no_scheme, '', $admin_root );
```

Comment:
Why strip the scheme and home URL from admin and rest URLs? Why not send the full URL?

This may also cause problems, particularly with the `admin_url` which is based on the `site_url` rather than the `home_url` (like `rest_url`).

Also, I think each one could probably be accomplished in 1 line rather than two by adding `$home_url_no_scheme` to the first array with the schemes.

Reply:
The proxy deals with these relatively, i.e. it only stores the relative path from the home URL to the REST API / WP Admin respectively.

The fact that `admin_url()` is based on `site_url()` is actually the reason why that parameter is even passed. For example if `home_url()` is `https://example.com` and `site_url()` is `https://example.com/wp`, then `$admin_root` will be `/wp/wp-admin/`.

Good point on simplifying to one line, will do that.